easybuild/tools/modules.py

```diff
@@ -386,14 +408,25 @@ def available(self, mod_name=None, extra_args=None):
             extra_args = []
         if mod_name is None:
             mod_name = ''
-        args = ['avail'] + extra_args + [mod_name]
-        mods = self.run_module(*args)
 
-        # sort list of modules in alphabetical order
-        mods.sort(key=lambda m: m['mod_name'])
-        ans = nub([mod['mod_name'] for mod in mods])
+        # cache 'avail' calls without an argument, since these are particularly expensive...
+        key = ('MODULEPATH=%s' % os.environ.get('MODULEPATH', ''), ';'.join(extra_args))
+        if not mod_name and key in MODULE_AVAIL_CACHE:
```

Comment:
yes, that's also by far the most expensive and common call

also caching argument-less avail calls would need more care w.r.t. invalidating the cache, etc.